server/src-lib/Hasura/RQL/DDL/Relationship.hs

```diff
@@ -153,46 +154,49 @@ createObjRelP2 (WithTable qt rd) = do
   return successMsg
 
 runCreateRemoteRelationship ::
-     (MonadTx m, CacheRM m) => CreateRemoteRelationship -> m EncJSON
-runCreateRemoteRelationship createRemoteRelationship = do
-  runCreateRemoteRelationshipP1 createRemoteRelationship
-  runCreateRemoteRelationshipP2 createRemoteRelationship
+     (MonadTx m, CacheRWM m) => RemoteRelationship -> m EncJSON
+runCreateRemoteRelationship remoteRelationship = do
+  remoteField <- runCreateRemoteRelationshipP1 remoteRelationship
+  runCreateRemoteRelationshipP2 remoteField
 
 runCreateRemoteRelationshipP1 ::
-     (MonadTx m, CacheRM m) => CreateRemoteRelationship -> m ()
-runCreateRemoteRelationshipP1 createRemoteRelationship = do
+     (MonadTx m, CacheRM m) => RemoteRelationship -> m RemoteField
+runCreateRemoteRelationshipP1 remoteRelationship = do
   sc <- askSchemaCache
   case HM.lookup
-         (ccrRemoteSchema createRemoteRelationship)
+         (rtrRemoteSchema remoteRelationship)
          (scRemoteResolvers sc) of
     Just {} -> do
       validation <-
-        getCreateRemoteRelationshipValidation createRemoteRelationship
+        getCreateRemoteRelationshipValidation remoteRelationship
       case validation of
         Left err -> throw400 RemoteSchemaError (T.pack (show err))
-        Right {} -> pure ()
+        Right remoteField -> pure remoteField
     Nothing -> throw400 RemoteSchemaError "No such remote schema"
 
 runCreateRemoteRelationshipP2 ::
-     (MonadTx m) => CreateRemoteRelationship -> m EncJSON
-runCreateRemoteRelationshipP2 createRemoteRelationship = do
-  liftTx (persistCreateRemoteRelationship createRemoteRelationship)
+     (MonadTx m, CacheRWM m) => RemoteField -> m EncJSON
+runCreateRemoteRelationshipP2 remoteField = do
+  liftTx (persistCreateRemoteRelationship (rmfRemoteRelationship remoteField))
+  addRemoteFieldToCache remoteField noDependencies
   pure successMsg
+  where
+    noDependencies = mempty
```

Comment:
We need to add 3 dependencies here:

1. Table dependency: RemoteRel depends on the table
2. Column dependencies: All the Tables columns are also a dependency for the RemoteRel
3. Remote Schema dependency: RemoteRel depends on the remote schema

1 and 2 can be handled like in the POC PR, 3) needs to be figured out as there is no remote schema dependency object as of today.

Reply:
So what do we need for (3)? Should I just add a constructor to [SchemaObjId](https://github.com/chrisdone/graphql-engine/blob/96f83b58130484dfd7c4b13cfdb969cc188a587d/server/src-lib/Hasura/RQL/Types/SchemaCacheTypes.hs#L26) and see the damage measured in compile warnings?

Reply:
Yes! Something like `SORemoteSchema !RemoteSchemaName` ?

Reply:
Great. I just started with the same thing. :ok_hand: